submitit/core/core.py

```diff
@@ -364,12 +359,21 @@ def _get_outcome_and_result(self) -> tp.Tuple[str, tp.Any]:
         while not self.paths.result_pickle.exists() and _time.time() - start_wait < timeout:
             _time.sleep(1)
         if not self.paths.result_pickle.exists():
+            message = [
+                f"Job {self.job_id} (task: {self.task_id}) with path {self.paths.result_pickle}",
+                f"has not produced any output (state: {self.state})",
+            ]
             log = self.stderr()
-            raise utils.UncompletedJobError(
-                f"Job {self.job_id} (task: {self.task_id}) with path {self.paths.result_pickle}\n"
-                f"has not produced any output (state: {self.state})\n"
-                f"Error stream produced:\n----------------------\n{log}"
-            )
+            if log:
+                message.extend(["Error stream produced:", "-" * 40, log])
+            elif self.paths.stdout.exists():
+                log = subprocess.check_output(["tail", "-40", str(self.paths.stdout)], encoding="utf-8")
+                message.extend(
+                    [f"No error stream produced. Look at stdout: {self.paths.stdout}", "-" * 40, log]
+                )
+            else:
+                message.append(f"No output/error stream produced ! Check: {self.paths.stdout}")
```

Comment:
The `check: {self.paths.stdout}` is a bit misleading, as this is only executed if the file does not exist.

Reply:
Normally the first thing that submitit does when starting is to log to this file. So I don't think this case can happen unless there is something very wrong with the filesystem. 
Maybe `check` is the wrong word, what about `stdout should have been at:` ?